addons/mail/static/src/core/common/navigable_list.xml

```diff
@@ -12,16 +12,19 @@
                 </div>
                 <t t-else="">
                     <div
-                        t-foreach="props.options" t-as="option" t-key="option_index"
+                        t-foreach="sortedOptions" t-as="option" t-key="option_index"
                         class="o-mail-NavigableList-item"
                         t-att-class="option.classList"
                         t-on-mouseenter="() => this.onOptionMouseEnter(option_index)"
                         t-on-click="(ev) => this.selectOption(ev, option_index)"
                     >
+                        <hr class="my-2" t-if="option.group != lastGroup and option_index != 0"/>
```

Comment:
t-if always at the beginning of the element. `<hr t-if="option.group != lastGroup and option_index != 0" class="my-2" />`